sonar-cxx-plugin/src/test/java/org/sonar/plugins/cxx/squid/CxxSquidSensorTest.java

```diff
@@ -19,200 +19,178 @@
+import org.sonar.plugins.cxx.CxxLanguage;
 
 public class CxxSquidSensorTest {
 
   private CxxSquidSensor sensor;
-  private SensorContext context;
   private Settings settings;
-  private FileSystem fs;
-  private Project project;
-  private ResourcePerspectives perspectives;
-  private Highlightable highlightable;
-  private Highlightable.HighlightingBuilder builder;
-
+  
   @Before
   public void setUp() {
     settings = new Settings();
-    context = mock(SensorContext.class);
-    perspectives = mock(ResourcePerspectives.class);
-    highlightable = mock(Highlightable.class);
-    builder = mock(Highlightable.HighlightingBuilder.class);
-
-    when(context.isIndexed(any(Resource.class), anyBoolean())).thenReturn(true); //@todo isIndexed: deprecated, see http://javadocs.sonarsource.org/4.5.2/apidocs/deprecated-list.html
-    when(perspectives.as(eq(Highlightable.class), any(InputFile.class))).thenReturn(highlightable);
-    when(highlightable.newHighlighting()).thenReturn(builder);
+    ActiveRules rules = mock(ActiveRules.class);
+    CheckFactory checkFactory = new CheckFactory(rules);
+    sensor = new CxxSquidSensor(settings, checkFactory, rules, null);    
   }
 
   @Test
-  public void testCollectingSquidMetrics() {
+  public void testCollectingSquidMetrics() throws IOException {
     File baseDir = TestUtils.loadResource("/org/sonar/plugins/cxx/codechunks-project");
-    setUpSensor(baseDir, Arrays.asList(new File(".")));
-
-    sensor.analyse(project, context);
-
-    verify(context).saveMeasure((InputFile) anyObject(), eq(CoreMetrics.FILES), eq(1.0));
-    verify(context).saveMeasure((InputFile) anyObject(), eq(CoreMetrics.LINES), eq(92.0));
-    verify(context).saveMeasure((InputFile) anyObject(), eq(CoreMetrics.NCLOC), eq(54.0));
-    verify(context).saveMeasure((InputFile) anyObject(), eq(CoreMetrics.STATEMENTS), eq(50.0));
-    verify(context).saveMeasure((InputFile) anyObject(), eq(CoreMetrics.FUNCTIONS), eq(7.0));
-    verify(context).saveMeasure((InputFile) anyObject(), eq(CoreMetrics.CLASSES), eq(0.0));
-    verify(context).saveMeasure((InputFile) anyObject(), eq(CoreMetrics.COMPLEXITY), eq(19.0));
-    verify(context).saveMeasure((InputFile) anyObject(), eq(CoreMetrics.COMMENT_LINES), eq(15.0));
+    
+    SensorContextTester context = SensorContextTester.create(baseDir);    
+    
+    String fileName = "code_chunks.cc";
+    String content = new String(Files.readAllBytes(new File(baseDir, fileName).toPath()), "UTF-8");
+    context.fileSystem().add(new DefaultInputFile("myProjectKey", fileName).initMetadata(content).setLanguage(CxxLanguage.KEY).setType(InputFile.Type.MAIN));    
+    sensor.execute(context);
+    Collection<Measure> measures = context.measures("myProjectKey:code_chunks.cc");
+            
+    assertThat(GetIntegerMeasureByKey(measures, CoreMetrics.FILES).value()).isEqualTo(1);
+    assertThat(GetIntegerMeasureByKey(measures, CoreMetrics.NCLOC).value()).isEqualTo(54);
```

Comment:
should, at least it shows correctly in web ui,. i think its not set for our internal testing, now where its actually done i dont know. likey done by the sonar-scanner itself or in server side. either way this metric is not be checked inside our unit tests. however it should be possible to verify in the integrations tests.